### PR TITLE
p2p candidate

### PR DESCRIPTION
**Pull request by coot**

### CODE REVIEW

File: cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs [hunk 1]
```diff
@@ -304,11 +312,11 @@ instance HasSeverityAnnotation (WithMuxBearer peer MuxTrace) where
...
     MuxTraceSDUReadTimeoutException -> Notice
     MuxTraceSDUWriteTimeoutException -> Notice
-    MuxTraceStartEagerly _ _ -> Debug
-    MuxTraceStartOnDemand _ _ -> Debug
-    MuxTraceStartedOnDemand _ _ -> Debug
-    MuxTraceTerminating {} -> Debug
+    MuxTraceStartEagerly _ _ -> Info
+    MuxTraceStartOnDemand _ _ -> Info
+    MuxTraceStartedOnDemand _ _ -> Info
     MuxTraceShutdown -> Debug
```
Comment (karknu) [inline cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs:372]: This doesn't match the commit message.

Comment (coot) [reply cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs:372]: Done; I fixed this after pushing, but I didn't wanted intervene the CI.

File: cardano-node/src/Cardano/Node/Configuration/POM.hs [hunk 2]
```diff
@@ -334,6 +418,13 @@ defaultPartialNodeConfiguration =
     , pncLogMetrics = mempty
     , pncTraceConfig = mempty
     , pncMaybeMempoolCapacityOverride = mempty
+    , pncProtocolIdleTimeout   = Last (Just 5)
+    , pncTimeWaitTimeout       = Last (Just 60)
+    , pncTargetNumberOfRootPeers        = Last (Just 5)
+    , pncTargetNumberOfKnownPeers       = Last (Just 5)
+    , pncTargetNumberOfEstablishedPeers = Last (Just 2)
+    , pncTargetNumberOfActivePeers      = Last (Just 1)
```
Comment (karknu) [inline cardano-node/src/Cardano/Node/Configuration/POM.hs]: Doesn't match the proposed defaults.

File: cardano-node/src/Cardano/Node/Configuration/TopologyP2P.hs [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+  toJSON top =
+    case top of
+      RealNodeTopology lrpg prp ul -> object [ "LocalRoots"         .= lrpg
+                                             , "PublicRoots"        .= prp
+                                             , "useLedgerAfterSlot" .= ul
+                                             ]
+
+-- | Read the `NetworkTopology` configuration from the specified file.
+-- While running a real protocol, this gives your node its own address and
+-- other remote peers it will attempt to connect to.
```
Comment (karknu) [inline cardano-node/src/Cardano/Node/Configuration/TopologyP2P.hs:176]: This is oddly formulated and incorrect. The node doesn't learn of its own address by reading the topology file. I think that may have been true in `cardano-sl`.

File: configuration/defaults/byron-mainnet/configuration.yaml [hunk 4]
```diff
@@ -200,6 +200,34 @@ TraceHandshake: False
 # Trace IP Subscription messages.
 TraceIpSubscription: True
 
+# Trace local root peers
+TraceLocalRootPeers: True
+
+# Trace public root peers
+TracePublicRootPeers: False
```
Comment (karknu) [inline configuration/defaults/byron-mainnet/configuration.yaml:207]: I think this should be enabled.

File: configuration/cardano/mainnet-p2p-toplogy.json [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+          ],
+          "advertise": false
+        },
+        "valency": 1
+      }
+    ]
+  }
+  , "PublicRoots": []
+}
+
```
Comment (karknu) [inline configuration/cardano/mainnet-p2p-toplogy.json:20]: This is wrong.
`relays-new.cardano-mainnet.iohk.io` is a public root peer.
```
{
  "LocalRoots": {
    "groups": [
      ]
  },
  "PublicRoots": [
    {
      "publicRoots": {
        "addrs": [
          {
            "addr": "relays-new.cardano-mainnet.iohk.io",
            "port": 3001
          }
        ],
        "advertise": false
      }
    }
  ]
}
```

File: cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs [hunk 6]
```diff
@@ -318,6 +343,48 @@ instance HasSeverityAnnotation (WithMuxBearer peer MuxTrace) where
...
+      TracePromoteWarmPeers      {} -> Info
+      TracePromoteWarmLocalPeers {} -> Info
+      TracePromoteWarmFailed     {} -> Error
+      TracePromoteWarmDone       {} -> Info
+      TraceDemoteWarmPeers       {} -> Info
+      TraceDemoteWarmFailed      {} -> Error
+      TraceDemoteWarmDone        {} -> Info
+      TraceDemoteHotPeers        {} -> Info
+      TraceDemoteLocalHotPeers   {} -> Info
+      TraceDemoteHotFailed       {} -> Error
```
[the remark this answers is not in the document]
Comment (coot) [reply cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs:408]: so only `TracePublicRootsFailure` is left as `Error`

File: cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs [hunk 7]
```diff
@@ -791,15 +816,19 @@ instance (HasHeader header, ConvertRawHash header)
                (case pt of
                   GenesisPoint -> "Genesis"
                   BlockPoint _ h -> renderHeaderHash (Proxy @header) h)
+             , "delay" .= delay
+             , "size" .= blockSize
```
[the remark this answers is not in the document]
Comment (coot) [reply cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs:1076]: Fixed

File: configuration/cardano/mainnet-p2p-toplogy.json (hunk 5 above)
Comment (coot) [reply configuration/cardano/mainnet-p2p-toplogy.json:20]: ok, I'll move it to `PublicRoots` then

Comment (coot) [reply configuration/cardano/mainnet-p2p-toplogy.json:20]: Done

**Review by karknu — APPROVED** [review]
LGTM

**Comment by coot** [comment]
Forced push onto `p2p-master`.

**Pull request closed without merging**